spyder_kernels/comms/commbase.py

```diff
@@ -0,0 +1,344 @@
+
+        Parameters
+        ----------
+
+        spyder_msg_type: str
+            The spyder message type
+        content: dict
+            The (JSONable) content of the message
+        data: any
+            Any object that is serializable by cloudpickle (should be most
+            things). Will arrive as cloudpickled bytes in `.buffers[0]`.
+        """
+        if not self.is_open():
+            raise RuntimeError("The comm is not connected.")
+        import cloudpickle
+        msg_dict = {
+            'spyder_msg_type': spyder_msg_type,
+            'content': content,
+            }
+        buffers = [cloudpickle.dumps(data, protocol=PICKLE_PROTOCOL)]
+        self._comm.send(msg_dict, buffers=buffers)
+
+    def remote_call(self, **settings):
+        """Get a handler for remote calls."""
+        return RemoteCallFactory(self, **settings)
+
+    # ---- Private -----
+    def _register_comm(self, comm):
+        """
+        Open a new comm to the kernel.
+        """
+        comm.on_msg(self._comm_message)
+        comm.on_close(self._comm_close)
+        self._comm = comm
+
+    def _comm_close(self, msg):
+        """Close comm."""
+        self._comm = None
+
+    def _comm_message(self, msg):
+        """
+        Handle internal spyder messages.
+        """
+        # Load the buffer. Only one is supported.
+        try:
+            if PY2:
+                buffer = cloudpickle.loads(msg['buffers'][0])
+            else:
+                buffer = cloudpickle.loads(bytes(msg['buffers'][0]))
+            load_exception = None
+
+        except Exception as e:
+            load_exception = e
+            buffer = None
+
+        # Get message dict
+        msg_dict = msg['content']['data']
+
+        spyder_msg_type = msg_dict['spyder_msg_type']
+
+        if spyder_msg_type in self._message_handlers:
+            self._message_handlers[spyder_msg_type](
+                msg_dict, buffer, load_exception)
+            return
+
+        logger.debug("No such spyder message type: %s" % spyder_msg_type)
+
+    def _handle_remote_call(self, msg_dict, buffer, load_exception):
+        """Handle a remote call."""
+        if load_exception:
+            raise load_exception
+        try:
+            return_value = self._remote_callback(
+                    buffer['call_name'],
+                    buffer['call_args'],
+                    buffer['call_kwargs'])
+            self._set_call_return_value(buffer, return_value)
+        except Exception as e:
+            self._set_call_return_value(buffer, e, exception=True)
+
+    def _remote_callback(self, call_name, call_args, call_kwargs):
+        """Call the callback function for the remote call."""
+        if call_name in self._remote_call_handlers:
+            return self._remote_call_handlers[call_name](
+                *call_args, **call_kwargs)
+
+        raise RuntimeError("No such spyder call type: %s" % call_name)
```

Comment:
Should this be a warning?

Reply:
This is triggered if an unregistered function is called. It could be that only blocking calls should be an error, and otherwise this is only a warning?